### PR TITLE
Scoring System Butterfly

### PR DESCRIPTION
**Pull request by lomky**
A series of improvements and revisions for the metrics system. 

## Scoring Changes

**Contributor**  
We decided that you don't need a reference when assessing that a person/org contributed to a publication.

Changes:
  - Contributor Acceptable: 
    - Old: `person + reference` OR `org + reference`
    - New: `Person + Role Type` OR `Org  + Role Type.`
  - Contributor Good: 
    - Old: `Person + Org + Reference`
    - New: `Person + Org + Role Type.`

## Script FIxes

  - Scores no longer stick on skipped score levels.
    - Persons, Instruments, and Model Runs may have had inflated scores prior.
  - Contributor subkeys now properly handled
    - Allows a Person score of 2. 
    - Would have been invisible due to sticky skipped bug for person score of 3.
  - Persons & Organizations no longer think they have "Contributors"
    - Contributors under these objects are _very different_ than the one under a publication.
    - Only in effect when running on a Person or Org directly.
  - Contributor & sub-Reference component URIs specified in a safer manner. 

## Misc Changes

  - Better verbose output for debugging

### CODE REVIEW

**Comment by lomky** [comment]
Do we have any more thoughts on score changes? 

I'm especially interested in anything @amruelama or @R-Aniekwu noticed while running against the old reports. Anything scoring higher or lower than you feel it should have?

**Comment by lomky** [comment]
Going to rename `depth` to `ref_depth` to better indicate what it affects. Setting its default to 0 as there is no benefit to the depth of 1+ until we have a rolled up scoring mechanism.

File: generate_resource_scores.pl [hunk 1]
```diff
@@ -73,7 +74,7 @@ =head1 EXAMPLES
   --resouce report/usgcrp-climate-human-health-assessment-2016/chapter/extreme-events \
```
Comment (amruelama) [inline generate_resource_scores.pl]: Fix typo 'resouce'

File: generate_resource_scores.pl [hunk 2]
```diff
@@ -43,9 +43,10 @@ =head1 OPTIONS
 
 File containing the component relationships. Default config/components.yaml
 
-=item B<--depth>
+=item B<--depth_references>
 
-How many references deep to process. Default 1.
+When encountering a reference, we will look at the child publication and all of
+its componenets and contributors this many times. Default 0.
```
Comment (amruelama) [inline generate_resource_scores.pl]: Fix typo 'componenets'

**Review by amruelama — APPROVED** [review]

**Pull request merged**